src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/scalable/VariantAnnotationWalker.java

```diff
@@ -0,0 +1,461 @@
+)
+@DocumentedFeature
+public class VariantAnnotationWalker extends MultiVariantWalker {
+
+    private static final String TRAINING_RESOURCE_LABEL = "training";
+    private static final String TRUTH_RESOURCE_LABEL = "truth";
+
+    private static final String ANNOTATIONS_HDF5_SUFFIX = ".annot.hdf5";
+    private static final String DEFAULT_VCF_SUFFIX = ".vcf";
+
+    private static final int DEFAULT_CHUNK_DIVISOR = 16;
+    private static final int DEFAULT_MAXIMUM_CHUNK_SIZE = HDF5Utils.MAX_NUMBER_OF_VALUES_PER_HDF5_MATRIX / DEFAULT_CHUNK_DIVISOR;
+
+    private static final String SCORE_KEY = GATKVCFConstants.VQS_LOD_KEY;
+    private static final String DUMMY_ALLELE = "<VQSR>";
+
+    @Argument(
+            fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+            shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+            doc = "Output prefix.")
+    String outputPrefix;
+
+    /**
+     * Any set of VCF files to use as lists of training or truth sites.
+     * Training - The program builds the model using input variants that overlap with these training sites.
+     * Truth - The program uses these truth sites to determine where to set the cutoff in sensitivity.
+     */
+    @Argument(
+            fullName = StandardArgumentDefinitions.RESOURCE_LONG_NAME,
+            doc = "") // TODO
+    private List<FeatureInput<VariantContext>> resources = new ArrayList<>(10);
+
+    @Argument(
+            fullName = "mode",
+            shortName = "mode",
+            doc = "Variant type to extract")
+    public VariantTypeMode mode = VariantTypeMode.SNP;
+
+    /**
+     * Extract per-allele annotations.
+     * Annotations should be specified using their full names with AS_ prefix.
+     * Non-allele-specific (scalar) annotations will be applied to all alleles.
+     */
+    @Argument(
+            fullName = "use-allele-specific-annotations",
+            shortName = "AS",
+            doc = "If specified, attempt to use the allele-specific versions of the specified annotations.",
+            optional = true)
+    public boolean useASannotations = false;
+
+    /**
+     * See the input VCF file's INFO field for a list of all available annotations.
+     */
+    @Argument(
+            fullName = "use-annotation",
+            shortName = "an",
+            doc = "The names of the annotations to extract.",
+            minElements = 1)
+    private List<String> useAnnotations = new ArrayList<>();
+
+    @Argument(
+            fullName = "ignore-filter",
+            doc = "If specified, use variants marked as filtered by the specified filter name in the input VCF file.",
+            optional = true)
+    private List<String> ignoreInputFilters = new ArrayList<>();
+
+    @Argument(
+            fullName = "ignore-all-filters",
+            doc = "If specified, ignore all input filters.",
+            optional = true)
+    private boolean ignoreAllFilters = false;
+
+    @Advanced
+    @Argument(
```

Comment:
I changed this to trust as the default (since that's recommended by all Best Practices/tutorials/etc. that I've seen) and added a comment about increased runtime if we choose not to trust. Not really sure about the impact/necessity, so I added a TODO in the code to document.